invoice/src/test/java/org/killbill/billing/invoice/api/user/TestDefaultInvoiceUserApi.java

```diff
@@ -418,6 +500,132 @@ public void testVoidInvoiceThatIsPaid() throws Exception {
+        final InvoiceItem itemCredit = itemCredits.get(0);
+
+        // No credit as invoice is still in DRAFT
+        final BigDecimal accountBalance = invoiceUserApi.getAccountBalance(accountId, callContext);
+        final BigDecimal accountCBA = invoiceUserApi.getAccountCBA(accountId, callContext);
+        Assert.assertEquals(accountBalance.compareTo(BigDecimal.ZERO), 0);
+        Assert.assertEquals(accountCBA.compareTo(BigDecimal.ZERO), 0);
+
+        // Create new invoice (COMMITTED)
+        final InvoiceItem externalCharge2 = new ExternalChargeInvoiceItem(null, accountId, null, "description", clock.getUTCToday(), clock.getUTCToday(), new BigDecimal("50.00"), accountCurrency, null);
+        final List<InvoiceItem> items2 =  invoiceUserApi.insertExternalCharges(accountId, clock.getUTCToday(), ImmutableList.<InvoiceItem>of(externalCharge2), true, null, callContext);
+        assertEquals(items2.size(), 1);
+        final Invoice invoice2 = invoiceUserApi.getInvoice(items2.get(0).getInvoiceId(), callContext);
+
+        // Verify balance and credit only reflect invoice2
+        final BigDecimal accountBalance2 = invoiceUserApi.getAccountBalance(accountId, callContext);
+        final BigDecimal accountCBA2 = invoiceUserApi.getAccountCBA(accountId, callContext);
+        Assert.assertEquals(accountBalance2.compareTo(new BigDecimal("50")), 0);
+        Assert.assertEquals(accountCBA2.compareTo(BigDecimal.ZERO), 0);
+
+        // Commit invoice 1
+        invoiceUserApi.commitInvoice(invoice1.getId(), callContext);
+
+        final BigDecimal accountBalance3 = invoiceUserApi.getAccountBalance(accountId, callContext);
+        final BigDecimal accountCBA3 = invoiceUserApi.getAccountCBA(accountId, callContext);
+        Assert.assertEquals(accountBalance3.compareTo(new BigDecimal("-50")), 0);
+        Assert.assertEquals(accountCBA3.compareTo(new BigDecimal("50")), 0);
+
+        // Void invoice1 where we initially generated the credit -> fails because there was credit generated on it
+        try {
+            invoiceUserApi.voidInvoice(invoice1.getId(), callContext);
+            Assert.fail("Should not allow to void invoice with credit");
+        } catch (final Exception ignore) {
+            // No check because of  https://github.com/killbill/killbill/issues/1501
+        }
 
+        Invoice latestInvoice1 = invoiceUserApi.getInvoice(items1.get(0).getInvoiceId(), callContext);
+        final InvoiceItem creditGenItem = Iterables.tryFind(latestInvoice1.getInvoiceItems(), new Predicate<InvoiceItem>() {
+            @Override
+            public boolean apply(final InvoiceItem invoiceItem) {
+                return InvoiceItemType.CBA_ADJ == invoiceItem.getInvoiceItemType() && invoiceItem.getAmount().compareTo(BigDecimal.ZERO) > 0;
+            }
+        }).get();
+
+        // Delete the credit generation
+        invoiceUserApi.deleteCBA(accountId, invoice1.getId(), creditGenItem.getId(), callContext);
```

Comment:
Out of curiosity, do we allow item adjusting these credit items via API?

Reply:
Nope we don't